src/v/serde/parquet/metadata.h

```diff
@@ -0,0 +1,380 @@
+    int64_t total_compressed_size;
+
+    /** Optional key/value metadata **/
+    std::vector<std::pair<ss::sstring, ss::sstring>> key_value_metadata;
+
+    /** Byte offset from beginning of file to first data page **/
+    int64_t data_page_offset;
+
+    /** Byte offset from beginning of file to root index page **/
+    int64_t index_page_offset;
+
+    /** Byte offset from the beginning of file to first (only) dictionary
+       page **/
+    int64_t dictionary_page_offset;
+};
+
+struct column_chunk {
+    /** File where column data is stored.  If not set, assumed to be same file
+     *as metadata.  This path is relative to the current file.
+     **/
+    std::optional<ss::sstring> file_path;
+
+    /** Column metadata for this chunk. Some writers may also replicate this at
+     *the location pointed to by file_path/file_offset. Note: while marked as
+     *optional, this field is in fact required by most major Parquet
+     *implementations. As such, writers MUST populate this field.
+     **/
+    column_meta_data meta_data;
+
+    /** File offset of ColumnChunk's OffsetIndex **/
+    int64_t offset_index_offset = 0;
+
+    /** Size of ColumnChunk's OffsetIndex, in bytes **/
+    int32_t offset_index_length = 0;
+
+    /** File offset of ColumnChunk's ColumnIndex **/
+    int64_t column_index_offset = 0;
+
+    /** Size of ColumnChunk's ColumnIndex, in bytes **/
+    int32_t column_index_length = 0;
+};
+
+/**
+ * Sort order within a RowGroup of a leaf column
+ */
+struct sorting_column {
+    /** The ordinal position of the column (in this row group) **/
+    int32_t column_idx;
+
+    /** If true, indicates this column is sorted in descending order. **/
+    bool descending;
+
+    /** If true, nulls will come before non-null values, otherwise,
+     * nulls go at the end. */
+    bool nulls_first;
+};
+
+struct row_group {
+    /** Metadata for each column chunk in this row group.
+     * This list must have the same order as the SchemaElement list in
+     * FileMetaData.
+     **/
+    chunked_vector<column_chunk> columns;
+
+    /** Total byte size of all the uncompressed column data in this row group
+     * **/
+    int64_t total_byte_size;
+
+    /** Number of rows in this row group **/
+    int64_t num_rows;
+
+    /** If set, specifies a sort ordering of the rows in this RowGroup.
+     * The sorting columns can be a subset of all the columns.
+     */
+    std::vector<sorting_column> sorting_columns;
+
+    /** Byte offset from beginning of file to first page (data or dictionary)
+     * in this row group **/
+    int64_t file_offset;
+};
+
+/**
+ * Description for file metadata
+ */
+struct file_metadata {
+    /** Version of this file */
+    int32_t version;
+    /**
+     * Parquet schema for this file.
+     */
+    chunked_vector<flattened_schema> schema;
+
+    /** Number of rows in this file **/
+    int64_t num_rows;
+
+    /** Row groups in this file */
+    chunked_vector<row_group> row_groups;
```

Comment:
Just curious is your line between chunked_vector and std::vector something to the effect of, size is application/datasize-bound -> chunked_vector, and size is spec-bound -> std::vector?

Reply:
Everything should probably just be chunked_vector - but yes, anything user/application defined is always chunked_vector, and std::vector only if we're very sure there is only going to be a small fixed amount of data (like I expect us not to write any parquet metadata actually, or very little if we do).